Pool/TwoFactorLogin.vb

```diff
@@ -0,0 +1,77 @@
+﻿Public Class TwoFactorLogin
+    Private _selectedEmail As String
+#Region "Vars/Properties"
+    Public Property SelectedEmail As String
+        Get
+            Return _selectedEmail
+        End Get
+        Set(value As String)
+            _selectedEmail = value
+        End Set
+    End Property
+
+    Dim twoFactorUser As New TwoFactorAuthenication
+    Dim code As Integer = 0
+#End Region
+
+    Private Sub TwoFactorAuth_Load(sender As Object, e As EventArgs) Handles MyBase.Load
+        twoFactorUser.User = SelectedEmail
+        SendCode()
+        Me.CenterToScreen()
+    End Sub
+
+    Private Sub btnSubmit_Click(sender As Object, e As EventArgs) Handles btnSubmit.Click
+        Dim userInputCode As Integer = 0
+        Integer.TryParse(tbCode.Text.Trim, userInputCode)
+        If userInputCode > 0 AndAlso twoFactorUser.Authenticate(userInputCode) Then
+            CurrentSession.isTwoFactorCode = True
+            ScoreTheme.LoadNextFormClose(Me, Home)
+            Me.Close()
+        Else
+            Dim incorrectAlert As DialogResult = MessageBox.Show($"The pin you entered is incorrect",
+                "Incorrect Pin", MessageBoxButtons.OK, MessageBoxIcon.Hand)
+        End If
+    End Sub
+
+    Private Sub btnResend_Click(sender As Object, e As EventArgs) Handles btnResend.Click
+        Dim previousCode As Integer = code
+        SendCode()
+
+        'Resend if it's the same code so how
+        'In case of a memory issue, catch it and default the pin
+        'Prompt for a retry
+        Try
+            If previousCode = code Then
+                While previousCode = code
+                    SendCode()
+                End While
+            End If
```

Comment:
We don't want to keep sending an email if we get lucky and generate the same one. Check the previous one before sending it again. If it's the same, keep generating it until it's different then send the email.

Reply:
This has been moved to Two Factor Class when generating the code. Duplicate pins against the last generated pin are checked before sending out any email.